src/Compilers/CSharp/Portable/Syntax/SyntaxFacts.cs

```diff
@@ -543,29 +544,54 @@ internal static bool HasAwaitOperations(SyntaxNode node)
                             });
         }
 
-        internal static bool IsNestedFunction(SyntaxNode child)
+        private static bool IsNestedFunction(SyntaxNode child)
+            => IsNestedFunction(child.Kind());
+
+        private static bool IsNestedFunction(SyntaxKind kind)
+            => kind is SyntaxKind.LocalFunctionStatement
+                or SyntaxKind.AnonymousMethodExpression
+                or SyntaxKind.SimpleLambdaExpression
+                or SyntaxKind.ParenthesizedLambdaExpression;
+
+        [PerformanceSensitive("", Constraint = "Use Green nodes for walking to avoid heavy allocations.")]
```

Comment:
#67198

Reply:
```suggestion
        [PerformanceSensitive("https://github.com/dotnet/roslyn/pull/66970", Constraint = "Use Green nodes for walking to avoid heavy allocations.")]
```

From [here](https://github.com/dotnet/roslyn/pull/67198/files#diff-fb3155646bc36c0bbb97405f51d8858056c02e1763b8deb6b159a5bbefaf74faR556)